OpenSmc.Ifrs17/ReportHub/ReportHubConfiguration.cs

```diff
@@ -1,52 +1,95 @@
-    //            .WithInitialization(ReportInit(config))));
-    //}
-
-    //public static Func<IMessageHub, CombinedWorkspaceState, CancellationToken, Task> ReportInit(MessageHubConfiguration config)
-    //{
-    //    return async (hub, workspace, cancellationToken) =>
-    //    {
-    //        await ReferenceDataHubConfiguration.RefDataInit(config, TemplateDimensions.Csv).Invoke(hub, workspace, cancellationToken);
-    //        await ParameterHubConfiguration.ParametersInit(config, TemplateParameter.Csv).Invoke(hub, workspace, cancellationToken);
-    //        await DataNodeHubConfiguration.DataNodeInit(config, "", null).Invoke(hub, workspace, cancellationToken);
-    //        await IfrsVariablesHubConfiguration.IfrsVarsInit(config, "").Invoke(hub, workspace, cancellationToken);
-
-    //        // TODO: WIP
-    //        var storage = new ReportStorage(workspace);
-    //        storage.Initialize((2020, 12), "CH", null, DataTypes.Constants.Enumerates.CurrencyType.Group);
-    //    };
-    //}
+        return configuration
+            .WithServices(services => services.AddSingleton<ScopeFactory>())
+            
+            .WithHostedHub(refDataAddress, config => config.ConfigureReferenceDataDictInit())
+            .WithHostedHub(dataNodeAddress, config => config.ConfigureDataNodeDataDictInit())
+            .WithHostedHub(parameterAddress, config => config.ConfigureParameterDataDictInit())
+
+            .WithHostedHub(ifrsVarAddress, config =>
+            {
+                var address = (IfrsVariableAddress)config.Address;
+                return config.ConfigureIfrsDataDictInit(address.Year, address.Month, address.ReportingNode, address.Scenario);
+            })
+            
+            .WithHostedHub(reportAddress, config => config
+                .AddData(data => data
+                    .FromConfigurableDataSource("ReportDataSource", ds => ds
+                        .WithType<ReportVariable>(t => t.WithKey(ReportVariableKey)))
+                    .FromHub(refDataAddress, ds => ds
+                        // TODO: complete this list
+                        .WithType<AmountType>().WithType<LineOfBusiness>())
+                    .FromHub(dataNodeAddress, ds => ds
+                        .WithType<InsurancePortfolio>().WithType<GroupOfInsuranceContract>()
+                        .WithType<ReinsurancePortfolio>().WithType<GroupOfReinsuranceContract>())
+                    .FromHub(parameterAddress, ds => ds 
+                        .WithType<ExchangeRate>().WithType<CreditDefaultRate>().WithType<PartnerRating>())
+                    .FromHub(ifrsVarAddress, ds => ds
+                        .WithType<IfrsVariable>())
+                    .AddCustomInitialization(ReportInit(config))))
+
+            .WithRoutes(route => route.RouteMessage<GetManyRequest<ReportVariable>>(_ => reportAddress));
+    }
+
+    public static Action<HubDataSource, ScopeFactory> ReportInit(MessageHubConfiguration config)
```

Comment:
What is HubDataSource?

Reply:
The data plugin has a HubDataSource inside to manage the data. And here I extract this in order to calculate ReportVariables.